src/lib/components/Modal/ModalBody.jsx

```diff
@@ -0,0 +1,58 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { withGlobalProps } from '../../provider';
+import { classNames } from '../../utils/classNames';
+import { isChildrenEmpty } from '../_helpers/isChildrenEmpty';
+import styles from './ModalBody.scss';
+
+export const ModalBody = ({
+  children,
+  id,
+  withScrollView,
+}) => {
+  if (isChildrenEmpty(children)) {
+    return null;
+  }
+
+  return (
+    <div
+      className={classNames(
+        styles.root,
+        withScrollView && styles.isRootScrollable,
+      )}
+      id={id}
+    >
+      {children}
+    </div>
+  );
+};
+
+ModalBody.defaultProps = {
+  children: null,
+  id: undefined,
+  withScrollView: false,
+};
+
+ModalBody.propTypes = {
+  /**
+   * Nested elements. Supported types are:
+   *
+   * * `ModalContent`
+   * * `ScrollView` (`withScrollView: true` must be set)
+   *
+   * At most one nested element is allowed. If none are provided nothing is rendered.
+   */
+  children: PropTypes.node,
+  /**
+   * ID of the root HTML element.
+   */
+  id: PropTypes.string,
+  /**
+   * If `true`, body is scroll-compatible and its child must be instance of `ScrollView`.
+   */
+  withScrollView: PropTypes.bool,
```

Comment:
Would it be possible to just check if the child is an instance of `ScrollView` or `ModalContent` via `React.Children.only()` method (https://reactjs.org/docs/react-api.html#reactchildren)?

Reply:
Yeah, it would. I was thinking and actually implementing this, but ... You can configure your own ScrollView or wrap it into some another component like MyCustomizedScrollView and it would stop working then. It is connected to the ongoing discussion above.

Reply:
> You can configure your own ScrollView or wrap it into some another component like MyCustomizedScrollView and it would stop working then.

I agree, but I dislike the `withScrollView` prop very much. If possible I would like to find a good solution to this as changing it later would be a BC change.

Some other non-great ideas how we could get rid of it.

1. We could check for presence of `ModalContent`. This component has very limited API so there is little need to wrap it. If the child is not `ModalContent` we now it is some version of `ScrollView` since nothing else is allowed.
2. We could check for `ScrollView` if `withScrollView` prop is undefined. That way the user does not have to set `withScrollView` if  she uses the `ScrollView` component unwrapped.

It would be ideal if we could modify the CSS to make it work regardless of weather it contains `ScrollView` or not. It would be much easier to modify `ScrollView` based on its parent via context then the other way around…

Reply:
I tried and I failed. We must always tell `ModalBody` what is responsible for scrolling, there is no other way. I'm suggesting more scrolling modes in #393 so it feels more like a feature than a workaround.

Reply:
Solved in #393